### DISCUSSION TITLE
AM2302 on an old ESP8266 Sonoff S20 - GPIO1 (tx) not working- RX (GPIO3) is working fine

### DISCUSSION THREAD

**Discussion by SimonFili** [arendst/Tasmota#15631] [Q&A] [answered]
Here's my setup.

Sonoff S20 (ESP8266) flashed with Tasmota.  
AM2302 connected to TX - GPIO1

The AM2302 is working fine after power ON, but not working after a soft RESET.

The AM2302 is entering a mode where the data line is always UP (3.3v).

If you disconnect the sensor, while the Sonoff is connected and reconnect the sensor, it works
Not working = null values, log showing (Timeout waiting for start signal high pulse)

I tried the following:
1) Different versions of firmware 10, 11
2) Recompiled with different timeout and delay in the code for DHT.ino 
3) Tried the same sensor on another ESP8266 with V11.x but on a GPIO not used for TX/RX, working, that confirmed the sensor was OK.  Power on or reset worked.

SOLVED:  I simply used the RX pin GPIO3 and AM2302 works at boot and soft reset.

I hope this will be useful for others.

**Comment by ascillato2** [comment, marked as answer]
Thanks for sharing your tests.

That issue is because the TX pin is sending hardware debug data at boot time. That can not be avoided and confuses the sensor making it to hung.

**Reply by SimonFili** [reply]
I would add that on the Tx/Rx pins of the Sonoff, we could use those pins for i2c sensors.  Just remember, the Tx "breaks" the One wire sensor.

**Reply by sfromis** [reply]
IIRC, using the tx/rx pins for I2C (at least for BME280) only works with tx as SCL and rx as SDA, not vice versa. As you cannot avoid tx pin activity at boot time, how a device deals with unsupported data depends a lot on details.

**Reply by SimonFili** [reply]
Just occurred to me... I understand the TX pin is busy and mess up the sensor, but why is it only happening with a "soft reset" and not on power up?

**Reply by sfromis** [reply]
No clear answer, but one possible part is that the sensor may need some boot time till able to get confused by "noise" on what should've been I2C.
